pages/reset-password/index.tsx

```diff
@@ -1,22 +1,153 @@
-import Input from "@/components/primitives/Form/Input";
 import Image from "next/image";
-import React, { useState } from "react";
-import { useRouter } from "next/navigation";
+import React, { useEffect, useState } from "react";
+import { useRouter, NextRouter } from "next/router";
 import Link from "next/link";
+import Input from "@/components/primitives/Form/Input";
+import Button from "@/components/primitives/Button/button";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { z, ZodError, ZodType } from "zod";
+
+// Define custom API error type
+type ApiErrorResponse = {
+  status: "Error";
+  message: string;
+};
+
+interface FormValues {
+  password: string;
+  confirmPassword: string;
+}
+
+type formData = {
+  password: string;
+  confirmPassword: string;
+};
 
-export default function ResetPassword() {
-  const router = useRouter();
-  const [values, setValues] = useState({
-    password: "",
-    confirmPassword: "",
+interface QueryParams {
+  token?: string;
+}
+
+// Define Zod schema for password and confirmPassword validation
+const passwordSchema: ZodType<formData> = z
+  .object({
+    password: z
+      .string()
+      .min(8, {
+        message: "Password must must not be less than 8 characters",
+      })
+      .refine((value) => /[!@#$%^&*(),.?":{}|<>]/.test(value), {
```

Comment:
Instead of doing this method of refine just to test regex, use the `.regex(put-your-regex-here)` method that zod provides. You will attach it to the string method like this z.string( ).regex( )